src/riak_kv_ts_newtype.erl

```diff
@@ -0,0 +1,200 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_kv_ts_newtype 
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(riak_kv_ts_newtype).
+-behaviour(gen_server).
+
+%% API.
+-export([start_link/0]).
+-export([new_type/1]).
+
+%% gen_server.
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([terminate/2]).
+-export([code_change/3]).
+
+-record(state, {
+}).
+
+-define(new_metadata_appeared,new_metadata_appeared).
```

Comment:
* Macro should be all caps
* Suggest renaming the atom to make more ops-friendly, perhaps  `bucket_type_changed_mid_compile`